pmm-app/src/pmm-qan/panel/components/Details/Table/TableContainer.hooks.ts

```diff
@@ -11,14 +11,16 @@ export const useTables = (example, explains, databaseType): any[] => {
     const getTables = async () => {
       setLoading(true);
       setTables([]);
-      if (databaseType === Databases.mysql && jsonExplain.value) {
-        const parsedJSON = JSON.parse(jsonExplain.value);
+      if (databaseType === Databases.mysql && example && jsonExplain) {
+        // eslint-disable-next-line global-require
+        const parser = require('sqlite-parser');
+        const ast = parser(example?.example);
 
-        const realTableName = [
-          get(parsedJSON, 'real_table_name'),
+        const tablesResult = [
+          get(ast, 'statement.0.from.source.name') || get(ast, 'statement.0.from.name'),
```

Comment:
do we need to check whether `statement` has non empty array? could be here the case if `statement` is empty or null?

Reply:
in general, if we get an example and jsonExplain we will have a statement with the proper query.